crates/nu-engine/src/documentation.rs

```diff
@@ -735,48 +781,36 @@ pub fn get_flags_section<F>(
 where
     F: FnMut(FormatterValue) -> String,
 {
+    if signature.named.is_empty() {
+        return "".to_owned();
+    }
+
     let help_section_name = &help_style.section_name;
     let help_subcolor_one = &help_style.subcolor_one;
     let help_subcolor_two = &help_style.subcolor_two;
 
     let mut long_desc = String::new();
     let _ = write!(long_desc, "\n{help_section_name}Flags{RESET}:\n");
-    for flag in &signature.named {
-        // Indentation
-        long_desc.push_str("  ");
-        // Short flag shown before long flag
-        if let Some(short) = flag.short {
-            let _ = write!(long_desc, "{help_subcolor_one}-{short}{RESET}");
-            if !flag.long.is_empty() {
-                let _ = write!(long_desc, "{DEFAULT_COLOR},{RESET} ");
-            }
-        }
-        if !flag.long.is_empty() {
-            let _ = write!(long_desc, "{help_subcolor_one}--{}{RESET}", flag.long);
-        }
-        if flag.required {
-            long_desc.push_str(" (required parameter)")
-        }
-        // Type/Syntax shape info
-        if let Some(arg) = &flag.arg {
-            let _ = write!(long_desc, " <{help_subcolor_two}{arg}{RESET}>");
-        }
-        if !flag.desc.is_empty() {
-            let _ = write!(
-                long_desc,
-                ": {}",
-                &formatter(FormatterValue::CodeString(&flag.desc))
-            );
-        }
-        if let Some(value) = &flag.default_value {
-            let _ = write!(
-                long_desc,
-                " (default: {})",
-                &formatter(FormatterValue::DefaultValue(value))
-            );
+
+    let mut flags = signature.named.iter();
+    let mut help_written = false;
+
+    // safe unwrap, since we know signature.named is not empty
+    let help = flags.next().unwrap();
```

Comment:
We could actually avoid this unwrap altogether if you move this line up to where you do the check at the beginning of the function, and instead do:
```rust
let Some(help) = flags.next() else {
    return "".into_owned()
}
```